### DISCUSSION TITLE
LASTVAL()

### DISCUSSION THREAD

**Discussion by dongri727** [serverpod/serverpod#776] [Q&A] [answered]
Could you tell me how to get lastval of insert item ?

**Comment by vlidholt** [comment, marked as answer]
The id of an inserted object is available in the `id` field of the object after the insert has completed.

**Comment by dongri727** [comment]
finally I wrote like this to make relational table

 Future<int> addPrincipal(Session session, Principal principal) async {
    await Principal.insert(session, principal);
    var principalLastVal = await session.db.query('SELECT LASTVAL()');
    return principalLastVal[0][0] as int;
  }

but it's not enough, still a work in progress
